internal/venice-common/src/main/java/com/linkedin/venice/views/MaterializedView.java

```diff
@@ -0,0 +1,88 @@
+  }
+
+  @Override
+  public Map<String, VeniceProperties> getTopicNamesAndConfigsForVersion(int version) {
+    VeniceProperties properties = new VeniceProperties(props);
+    String viewName = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_NAME.name());
+    return Collections.singletonMap(
+        Version.composeKafkaTopic(store.getName(), version) + VIEW_TOPIC_SEPARATOR + viewName
+            + MATERIALIZED_VIEW_TOPIC_SUFFIX,
+        properties);
+  }
+
+  /**
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITION_COUNT} is required to configure a new re-partition view.
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITIONER} is optional. The re-partition view will use the store level
+   * partitioner config if it's not specified in the view parameters.
+   * {@link ViewParameterKeys#MATERIALIZED_VIEW_PARTITIONER_PARAMS} is optional.
+   */
+  @Override
+  public void validateConfigs() {
+    super.validateConfigs();
+    String viewName = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_NAME.name());
+    if (viewName == null) {
+      throw new VeniceException(
+          String.format(MISSING_PARAMETER_MESSAGE, ViewParameterKeys.MATERIALIZED_VIEW_NAME.name()));
+    }
+    if (store.getViewConfigs().containsKey(viewName)) {
+      throw new VeniceException("A view config with the same view name already exist, view name: " + viewName);
+    }
+    String viewPartitioner = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_PARTITIONER.name());
+    if (viewPartitioner == null) {
+      viewPartitioner = store.getPartitionerConfig().getPartitionerClass();
+    }
+    try {
+      Class.forName(viewPartitioner);
+    } catch (ClassNotFoundException e) {
+      throw new VeniceException("Cannot find partitioner class: " + viewPartitioner);
+    }
+    String partitionCountString = viewParameters.get(ViewParameterKeys.MATERIALIZED_VIEW_PARTITION_COUNT.name());
+    int viewPartitionCount =
+        partitionCountString == null ? store.getPartitionCount() : Integer.parseInt(partitionCountString);
+    // A materialized view with the exact same partitioner and partition count as the store is not allwoed
+    if (store.getPartitionCount() == viewPartitionCount
+        && store.getPartitionerConfig().getPartitionerClass().equals(viewPartitioner)) {
+      throw new VeniceException(
+          "A materialized view with the same partitioner and partition count as the original store is not allowed!");
+    }
+    // Check if there is already a materialized view with identical configs
+    for (Map.Entry<String, ViewConfig> viewConfigEntries: store.getViewConfigs().entrySet()) {
+      ViewConfig viewConfig = viewConfigEntries.getValue();
+      if (viewConfig.getViewClassName().equals(MaterializedView.class.getCanonicalName())) {
+        String configPartitioner =
+            viewConfig.getViewParameters().get(ViewParameterKeys.MATERIALIZED_VIEW_PARTITIONER.name());
+        if (configPartitioner == null) {
```

Comment:
In theory, the configPartitioner should never be null as if it is absent at the view config registration time, we will backfill it by the store-level config, right?
We should always have a valid partitioner in the view config as the store-level partitioner can change in the future.

Reply:
Good call, let me add that filling action.